astra/src/main/proto/astra_search.proto

```diff
@@ -23,6 +23,23 @@ message SearchRequest {
   // The fully query object, to be used in the leaf nodes and parsed by OpenSearch
   string query = 8;
 
+  // An explicit list of fields to return per document (effectively an allowlist)
+  FieldInclusion include_fields = 9;
```

Comment:
As mentioned above, I think matching these to a single source option that contains both is likely a better option and more closely matches the original API.